datasafe-rest/src/main/java/de/adorsys/datasafe/rest/controller/UserController.java

```diff
@@ -0,0 +1,82 @@
+package de.adorsys.datasafe.rest.controller;
+
+import de.adorsys.datasafe.business.api.types.CreateUserPrivateProfile;
+import de.adorsys.datasafe.business.api.types.CreateUserPublicProfile;
+import de.adorsys.datasafe.business.api.types.UserID;
+import de.adorsys.datasafe.business.api.types.UserIDAuth;
+import de.adorsys.datasafe.business.api.types.keystore.ReadKeyPassword;
+import de.adorsys.datasafe.business.api.types.resource.*;
+import de.adorsys.datasafe.business.impl.service.DefaultDatasafeServices;
+import de.adorsys.datasafe.rest.config.DatasafeProperties;
+import de.adorsys.datasafe.rest.dto.UserCreateRequestDTO;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RestController()
+@RequestMapping(value = "/user", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+public class UserController {
+
+   private DatasafeProperties properties;
+
+    private static final String PRIVATE_COMPONENT = "private";
+    private static final String PRIVATE_FILES_COMPONENT = PRIVATE_COMPONENT + "/files";
+    private static final String INBOX_COMPONENT = "inbox";
+
+    private final DefaultDatasafeServices dataSafeService;
+
+    public UserController(DatasafeProperties properties, DefaultDatasafeServices dataSafeService) {
+        this.properties = properties;
+        this.dataSafeService = dataSafeService;
+    }
+
+    @PutMapping
+    @ApiOperation("Create user")
+    public void createUser(@RequestBody UserCreateRequestDTO requestDTO) {
+
+        ReadKeyPassword readKeyPassword = new ReadKeyPassword(requestDTO.getPassword());
+        UserIDAuth auth = new UserIDAuth(new UserID(requestDTO.getUserName()), readKeyPassword);
+
+        URI rootLocation = URI.create(properties.getSystemRoot());
+        rootLocation = rootLocation.resolve(requestDTO.getUserName() + "/");
+        URI inboxUri = rootLocation.resolve("./" + INBOX_COMPONENT + "/");
+
+        URI keyStoreUri = rootLocation.resolve("./" + PRIVATE_COMPONENT + "/keystore");
+
+        dataSafeService.userProfile().registerPublic(CreateUserPublicProfile.builder()
```

Comment:
Please check BasicFunctionalityTest, there register user is 
```
UserIDAuth auth = new UserIDAuth(new UserID(userName), new ReadKeyPassword("secure-password " + userName));

        rootLocation = rootLocation.resolve(userName + "/");

        URI keyStoreUri = rootLocation.resolve("./" + PRIVATE_COMPONENT + "/keystore");
        log.info("User's keystore location: {}", Log.secure(keyStoreUri));
        URI inboxUri = rootLocation.resolve("./" + INBOX_COMPONENT + "/");
        log.info("User's inbox location: {}", Log.secure(inboxUri));

        AbsoluteLocation<PublicResource> publicKeys = access(
                rootLocation.resolve("./" + PUBLIC_COMPONENT + "/keystore")
        );

        profileRegistrationService.registerPublic(CreateUserPublicProfile.builder()
                .id(auth.getUserID())
                .inbox(access(inboxUri))
                .publicKeys(publicKeys)
                .build()
        );

        URI filesUri = rootLocation.resolve("./" + PRIVATE_FILES_COMPONENT + "/");
        log.info("User's files location: {}", Log.secure(filesUri));

        profileRegistrationService.registerPrivate(CreateUserPrivateProfile.builder()
                .id(auth)
                .privateStorage(accessPrivate(filesUri))
                .keystore(accessPrivate(keyStoreUri))
                .inboxWithWriteAccess(accessPrivate(inboxUri))
                .documentVersionStorage(accessPrivate(rootLocation.resolve("./" + VERSION_COMPONENT + "/")))
                .publishPubKeysTo(publicKeys)
                .build()
        );

        log.info("Created user: {}", Log.secure(userName));
        return auth;
```

Reply:
You are missing 
.documentVersionStorage(accessPrivate(rootLocation.resolve("./" + VERSION_COMPONENT + "/")))
 .publishPubKeysTo(publicKeys)

Reply:
ok